### ISSUE TITLE
`DataViewerApp`

### ISSUE THREAD

**Issue by kangz12345** [snu-quiqcl/iquip#143]
### Feature you want to implement
`DataViewerApp` features:
1. Source selection: Choose the data source among the current experiment (realtime), the result in the artiq-master (remote), the local file (local).
2. Histogram plot and state discrimination threshold & data type (P1 population, average photon count, total photon count) configuration
3. Main plot.
4. Tools, e.g., sinusoidal fitting
Conceptual drawing (sorry for the bad quality ;)):
<img width="776" alt="image" src="https://github.com/snu-quiqcl/iquip/assets/8445906/83ca303d-cf68-4ff2-b876-5343eda54c97">

### How the feature is implemented
- [x] #144 
- [x] #219
- [ ] Remote dataset
- [ ] Local dataset
- [ ] Fitting tool

**Comment by kangz12345** [comment]
> ### Feature you want to implement
> `DataViewerApp` features:
> 
> 1. Source selection: Choose the data source among the current experiment (realtime), the result in the artiq-master (remote), the local file (local).
> 2. Histogram plot and state discrimination threshold & data type (P1 population, average photon count, total photon count) configuration
> 3. Main plot.
> 4. Tools, e.g., sinusoidal fitting
>    Conceptual drawing (sorry for the bad quality ;)):
> 
> <img alt="image" width="776" src="https://user-images.githubusercontent.com/8445906/268170090-83ca303d-cf68-4ff2-b876-5343eda54c97.png">
> ### How the feature is implemented
> * First, a minimal version will be implemented: #144

**Comment by BECATRUE** [comment]
I'm little confused about the difference between "remote" and "local".
As I understood, "remote" method is fetching the H5 file from the proxy server and showing the data like `VisualizerApp`.
Is it right?

**Comment by kangz12345** [comment]
Yes, correct! "remote" shows the result data from the artiq master, so it requires the RID of the desired experiment.
On the other hand, "local" is for showing data files in local (machine). This could allow the data viewer to be more versatile. It doesn't need the artiq master.
